### ISSUE TITLE
Does POSTMAN support eBay OAuth2.0 token?

### ISSUE THREAD

**Issue by GingerMoon** [a85/POSTMan-Chrome-Extension#3537]
## App Details:
Version 5.2.0
win / x86-32
Chrome 60.0.3112.113

## Issue Report:
Not able to get eBay OAuth2.0 token via POSTMAN.
Steps to reproduce the problem
[reproduce steps.xlsx](https://github.com/postmanlabs/postman-app-support/files/1310158/reproduce.steps.xlsx)

The document of eBay OAuth2.0:
https://developer.ebay.com/api-docs/static/oauth-user-token.html
Not very sure whether POSTMAN handles the code appended at the end of https://www.getpostman.com/oauth2/callback

**Comment by harryi3t** [comment]
@GingerMoon 
I checked their docs and found this
![image](https://user-images.githubusercontent.com/5207331/30538168-54c98b00-9c8a-11e7-92b8-ae8bebb06c3e.png)
Correct me if I am wrong here,
They require that the `redirect_uri` to be the `RU-name-value` (a string generated from their UI, for me it was `Harendra_Singh-Harendra-Postma-xxxxxxxx`) 

But according to the [RFC](https://tools.ietf.org/html/rfc6749#section-3.1.2) the `redirect_uri` MUST be an absolute URI

> The redirection endpoint URI MUST be an absolute URI as defined by
>    [RFC3986] Section 4.3

So postman will send the `redirect_uri` as `https://www.getpostman.com/oauth2/callback ` which is pre-configured
But their server is expecting the string they generated (the `RU-name-value`)

Thoughts?

**Comment by krich** [comment]
Hi Harry,

There's not much we can do about using our in-house generated RuName value for the **redirect_uri** value.

We issue the following redirect URL, with query parameters, to get the 3p-permissions, and I am wondering how to set up this redirect URL in Postman:

`/* URL redirects a user to the apps Grant Application Access page */
  https://signin.sandbox.ebay.com/authorize?
    client_id=<your-client-id-value>&
    redirect_uri=<your-RuName-value>&
    response_type=code&
    state=<client-supplied-state-value>&
    scope=<list-of-scopes>`

Thanks in advance for your help.
   :- Kelly

**Comment by GingerMoon** [comment]
@harryi3t 
Hi Harry, thank you very much for looking into this issue, I really appreciate it.
"So postman will send the redirect_uri as https://www.getpostman.com/oauth2/callback which is pre-configured
But their server is expecting the string they generated (the RU-name-value)"

You are right that eBay's server is expecting the RU-name-value string. Actually, the according to this string, eBay server can find the redirect URL. As you can see from the attachment, the redirect URL is configured to the https://www.getpostman.com/oauth2/callback.

By the way, I would appreciate if you could share the flow/ sequence of the requests the postman sends, including how postman construct the request URL. 
For example, as you can see in my attachment, I have to append client id to the authorize URL because eBay server is expecting the client id but the postman does append it for me although there has already been a client id field been input in the postman.

To put it simply, I want to know what postman does according to the fields I input for OAuth2 fields.
For example, maybe postman will need client id and secret id for generating the HTTP header Authorization value (encoded Basic ...client id..secret id) .

**Comment by harryi3t** [comment]
@GingerMoon @krich 
We are making some UI changes for auths in Postman 5.3
As part of this, we are making some changes to the Oauth-2 modal as well.

I have tested locally that if we make the callback URL an input box (right now it's fixed to ` https://www.getpostman.com/oauth2/callback`) and put the RuName string in it, then it works!

The following works for me (local build)
![image](https://user-images.githubusercontent.com/5207331/30848501-f10d5b80-a2bc-11e7-84c2-1674fd449544.png)

This will be released with Postman 5.3 (around end of next week)

**Comment by GingerMoon** [comment]
@harryi3t Thank you very much! Looking forward to this version!

**Comment by krich** [comment]
@harryi3t, Thanks for this great news!

By any chance is there time frame for when Postman 5.3 is expected to be released?

**Comment by harryi3t** [comment]
Yes! it should be released  around end of the next week

**Comment by krich** [comment]
@harryi3t 
Hello Harry!
  I picked up the latest version of Postman and see the new UI for OAuth 2.0, GREAT!!!

However, I am having a little trouble getting my access token. I have completed the new GET NEW ACCESS TOKEN form, and click Request Token. 

The call-back works, and I am able to log into the Grant Access page and grant access. However, nothing happens after that. I don't get an Authorization Code to plug into the Authorization Code Grant request. Am I doing something wrong, or is there a process I need to follow?

Thanks for any help with this.
   :- Kelly

**Comment by harryi3t** [comment]
@krich Can you try these steps
- Restart the app (Quit and Reopen on mac)
- Make sure you send the client credentials in the header and not in body. For this choose `Send as basic auth header` from the `Client Authentication` dropdown

If it does now work, please see this small video that I recorded.
https://youtu.be/yzkE5I9ysBg

If you still have issues, you can contact me on our slack community channel.
https://www.getpostman.com/slack-invite

**Comment by SnehaPuthoor** [comment]
Hey @harryi3t , I am not familiar with postman param n authorisation settings.. could you please help me to explain what all the fields to be filled while I am trying to authenticate eBay api..
I am trying to check GET https://auth.eBay.com/oauth2/authorise


Really appreciate your help on this, please..

Thanks.

**Comment by GreatGleb** [comment]
> 

Hello, I did like this in my Postman, but I have error 
{"error_id":"unauthorized_client","error_description":"The OAuth client was not found.","http_status_code":401}
